internal/db/sshkeys.go

```diff
@@ -0,0 +1,118 @@
+// Copyright 2025 Canonical.
+package db
+
+import (
+	"context"
+	"fmt"
+
+	gossh "golang.org/x/crypto/ssh"
+
+	"github.com/canonical/jimm/v3/internal/dbmodel"
+	"github.com/canonical/jimm/v3/internal/errors"
+	"github.com/canonical/jimm/v3/internal/servermon"
+)
+
+// AddSSHKey adds a new SSH key.
+func (d *Database) AddSSHKey(ctx context.Context, sshKey *dbmodel.SSHKey) (err error) {
+	const op = errors.Op("db.AddSSHKey")
+	if err := d.ready(); err != nil {
+		return errors.E(op, err)
+	}
+
+	durationObserver := servermon.DurationObserver(servermon.DBQueryDurationHistogram, string(op))
+	defer durationObserver()
+	defer servermon.ErrorCounter(servermon.DBQueryErrorCount, &err, string(op))
+
+	if err := d.DB.WithContext(ctx).Create(sshKey).Error; err != nil {
+		return errors.E(op, dbError(err))
+	}
+	return nil
+}
+
+// RemoveSSHKeyByFingerprint removes a user's ssh key identified by its fingerprint.
+func (d *Database) RemoveSSHKeyByFingerprint(ctx context.Context, identityName string, fingerprint string) (err error) {
+	const op = errors.Op("db.RemoveSSHKeyByFingerprint")
+
+	if err := d.ready(); err != nil {
+		return errors.E(op, err)
+	}
+
+	durationObserver := servermon.DurationObserver(servermon.DBQueryDurationHistogram, string(op))
+	defer durationObserver()
+	defer servermon.ErrorCounter(servermon.DBQueryErrorCount, &err, string(op))
+
+	var keys []dbmodel.SSHKey
+	if err := d.DB.Where("identity_name = ?", identityName).Find(&keys).Error; err != nil {
+		return errors.E(op, dbError(err))
+	}
+
+	// It is expected that we only have 1 key that matches this fingerprint
+	// because of the unique constraint between users and public keys.
+	for _, key := range keys {
+		fp, err := calculateSSHFingerprint(key.PublicKey)
```

Comment:
ah, yes, i think we should be storing the fingerprint.. then this method would be much simpler and fingerprint is a static piece of data that never changes for a given key

Reply:
We can, I just initially took the route of not storing it since I figured users would have a small number of keys. I'll let others chime in then see about changing it.

Reply:
Changed it to store the fingerprint.